app/hooks/device.ts

```diff
@@ -43,23 +45,54 @@ export function useIsTablet() {
     return Device.IS_TABLET && !isSplitView;
 }
 
-export function useKeyboardHeight() {
+export function useKeyboardHeight(keyboardTracker?: React.RefObject<KeyboardTrackingViewRef>) {
     const [keyboardHeight, setKeyboardHeight] = useState(0);
+    const updateTimeout = useRef<NodeJS.Timeout | null>(null);
+    const insets = useSafeAreaInsets();
+
+    // This is a magic number. With tracking view, to properly get the final position, this had to be added.
+    const KEYBOARD_TRACKINGVIEW_SEPARATION = 4;
+
+    const updateValue = (v: number) => {
+        if (updateTimeout.current != null) {
+            clearTimeout(updateTimeout.current);
+            updateTimeout.current = null;
+        }
+        updateTimeout.current = setTimeout(() => {
+            setKeyboardHeight(v);
+            updateTimeout.current = null;
+        }, 200);
+    };
 
     useEffect(() => {
-        const show = Keyboard.addListener('keyboardWillShow', (event) => {
-            setKeyboardHeight(event.endCoordinates.height);
+        const show = Keyboard.addListener(Platform.select({ios: 'keyboardWillShow', default: 'keyboardDidShow'}), (event) => {
+            if (keyboardTracker?.current) {
+                // eslint-disable-next-line max-nested-callbacks
+                keyboardTracker.current.getNativeProps().then((x) => {
+                    if (x.keyboardHeight) {
+                        updateValue((x.trackingViewHeight + x.keyboardHeight) - KEYBOARD_TRACKINGVIEW_SEPARATION);
+                    } else {
+                        updateValue((x.trackingViewHeight + insets.bottom) - KEYBOARD_TRACKINGVIEW_SEPARATION);
+                    }
+                });
+            } else {
+                updateValue(event.endCoordinates.height);
+            }
```

Comment:
let's move this to a function in order to remove the eslint disable line or you can make the listener async.. both options work